src/lib/shady-render.ts

```diff
@@ -12,19 +12,29 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
-import {render as baseRender, Template, templateCaches, TemplateResult} from '../lit-html.js';
+import {removeNodes, Template, templateCaches, TemplateContainer, TemplateInstance, TemplateResult} from '../lit-html.js';
+
+import {insertNodeIntoTemplate, removeNodesFromTemplate} from './modify-template.js';
 
 export {html, svg, TemplateResult} from '../lit-html.js';
 
 declare global {
   interface Window {
     ShadyCSS: any;
   }
+  class ShadowRoot {}
 }
 
+const getTemplateKey = (type: string, scopeName: string) =>
```

Comment:
Consider a different name: `templateKey` sounds like it'd be based on the template contents... this is a key used to look up a cache.

Reply:
done